### ISSUE TITLE
Generate beautiful DMG files on CI

### ISSUE THREAD

**Issue by lucasfernog** [tauri-apps/tauri#1731]
A limitation of the bundler is that it uses [applescript](https://github.com/tauri-apps/tauri/blob/e37e187d4a3c7568463c2546099d03dd5a314f40/tooling/bundler/src/bundle/macos/templates/dmg/bundle_dmg#L328) to set the window size, icon position etc (so the DMG looks good). It doesn't work on CI because it accesses the Finder, which prompts for access.

The node-appdmg library creates a .DS_Store file instead of using applescript: https://github.com/LinusU/node-appdmg/blob/e1394d24b1be7799a3949ad0686cd9abb64339ee/lib/appdmg.js#L359

Maybe we could do the same? We would need to port [node-ds-store](https://github.com/LinusU/node-ds-store) to Rust.

**Comment by lucasfernog** [comment]
I'm considering using a fixed .DS_Store file instead of writing one at build time. This works right now because we don't expose configuration of the opened window / icons, but it would only work if document that the app icon must be named "icon.icns", which is the default name.

**Comment by nothingismagick** [comment]
We can do that naming change for sure

**Comment by lucasfernog** [comment]
Writing it right now. We can revisit this issue when someone requests customization of the DMG.

**Comment by lucasfernog** [comment]
Hmm the problem is actually the application name, not the icon name.

**Comment by lucasfernog** [comment]
Seems like we'll need the .DS_Store writer :/

**Comment by lorenzolewis** [comment]
I don't think this solve the whole problem, but maybe this could help with bits of it? https://www.unix.com/man-page/osx/1/productbuild/

**Comment by hjmallon** [comment]
If you own your own macOS CI (i.e. have access to the screen to allow the popup) then you can set `CI=false` in the job and enable DMG decoration.

**Comment by FabianLars** [comment]
I'm not sure if something changed but we've got a few reports of `CI=false` working in github ci too: https://github.com/tauri-apps/tauri-action/issues/740
Considering that i didn't see anything positive in the create-dmg i wouldn't jump the gun and change tauri though, just wanted to mention it here since this issue is now linked in the docs.

Also, i'm considering updating the tauri-action examples to set CI=false by default with a comment telling them when to remove that 🤔

**Comment by bes** [comment]
Any good workarounds for beautiful DMGs in macOS CI in 2026 ❤️ ?

**Comment by FabianLars** [comment]
Only what I said above, use GitHub ci (or maybe any CI that uses a full desktop instance and isn't running headless or whatever it's called).
Tauri-action will enable this automatically but for other places you can use the `TAURI_BUNDLER_DMG_IGNORE_CI=true` env var so you don't have to touch the CI var that may be read elsewhere.

But yeah, nothing changed so this won't work everywhere.

**Comment by bes** [comment]
Thank you for replying! I have a mac mini running in a cupboard but I can't get it to work. Maybe the lockscreen needs to be unlocked, but I don't want to leave unlocked computers around so I guess I'll have to wait until it's solved some other way.

**Comment by FabianLars** [comment]
any errors or just ugly dmgs? Can you share logs you get when building with `--verbose` flag?
Also, we're basically using https://github.com/create-dmg/create-dmg so maybe check if that works / or if any issues sound familiar.

a different solution will take time thanks to apple being apple.

**Comment by bes** [comment]
No errors, I just get a very basic DMG. When I build the DMG locally it looks much nicer :)

**Comment by bes** [comment]
I think my issue is that the tauri builder is running in a gitlab-runner which is a background process on my mac, so it never gets a chance to ask for the Automation grant. If I can figure out how to get the gitlab-runner binary to run the DMG packaging step as a "real ui app" then I think I can grant that one the permission.
